### PR TITLE
Fixing #10341 by directly using the camera space coordinate for

### PR DESCRIPTION
**Pull request by kevineschmidt**
Fixing #10341 by directly using the camera space coordinate for distance from camera, rather than using the approximation gl_FragCoord.z / gl_FragCoord.w

This did not seem to have a visual effect on any current examples, but the example attached to the original case illustrates it

### CODE REVIEW

File: src/renderers/shaders/ShaderChunk/fog_pars_vertex.glsl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#ifdef USE_FOG
+  varying float fogDepth;
+#endif
```
Comment (WestLangley) [inline src/renderers/shaders/ShaderChunk/fog_pars_vertex.glsl]: newline -- here and elsewhere

**Comment by WestLangley** [comment]
>This did not seem to have a visual effect on any current examples.

This looks good. However, I expect you are going to have to search the `examples/js` directory for the uses of `fog_fragment` and update those files too.

File: src/renderers/shaders/ShaderChunk/fog_pars_vertex.glsl (hunk 1 above)
Comment (WestLangley) [inline src/renderers/shaders/ShaderChunk/fog_pars_vertex.glsl]: I was referring to the lack of a new line at end of your glsl files... but do appreciate the improved formatting, though. :)

**Comment by kevineschmidt** [comment]
@WestLangley  - I changed the examples in examples/js. Suggestions on how to methodically test?

**Comment by mrdoob** [comment]
Thanks!

File: src/renderers/shaders/ShaderChunk/fog_fragment.glsl [hunk 2]
```diff
@@ -1,22 +1,12 @@
...
-	#else
-
-		float depth = gl_FragCoord.z / gl_FragCoord.w;
-
-	#endif
-
 	#ifdef FOG_EXP2
 
-		float fogFactor = whiteCompliment( exp2( - fogDensity * fogDensity * depth * depth * LOG2 ) );
+		float fogFactor = whiteCompliment( exp2( - fogDensity * fogDensity * fogDepth * fogDepth * LOG2 ) );
```
Comment (EliasHasle) [inline src/renderers/shaders/ShaderChunk/fog_fragment.glsl]: Is exp2(something*LOG2) intended here? exp(something*LOG2) would give exp2(something), if exp2 is the base 2 exponential and LOG2 is the logarithm of 2. exp2(something*LOG2) then makes the exponentiation base, assuming something is the argument, equal to exp(log(2)*log(2)).

I also wonder why the density and depth are squared. What is the physical analogy?

Comment (mrdoob) [reply src/renderers/shaders/ShaderChunk/fog_fragment.glsl]: Would you like opening a new issue to discuss this?

Comment (EliasHasle) [reply src/renderers/shaders/ShaderChunk/fog_fragment.glsl]: Yes, I think so (but not right now).

Consulting the source, `LOG2` is not the logarithm of 2, but the inverse of that. That means it converts a natural logarithm into a base 2 logarithm. So `exp2(something*LOG2)` is just a complicated way of writing `exp(something)`.

Comment (EliasHasle) [reply src/renderers/shaders/ShaderChunk/fog_fragment.glsl]: I am opening an issue right now...

Comment (EliasHasle) [reply src/renderers/shaders/ShaderChunk/fog_fragment.glsl]: #17264

**Pull request merged**